File: generated-sources/java-vertx/mojang-authentication/src/main/java/org/openapitools/server/api/verticle/DefaultApi.java
package org.openapitools.server.api.verticle;

import org.openapitools.server.api.model.AccessKeys;
import org.openapitools.server.api.model.Authentication;
import org.openapitools.server.api.model.AuthenticationRequest;
import org.openapitools.server.api.model.Error;
import org.openapitools.server.api.MainApiException;
import org.openapitools.server.api.model.RefreshRequest;
import org.openapitools.server.api.model.RefreshResponse;
import org.openapitools.server.api.model.UsernamePassword;

import io.vertx.core.AsyncResult;
import io.vertx.core.Handler;

import java.util.List;
import java.util.Map;

public interface DefaultApi  {
    //authenticate
    void authenticate(AuthenticationRequest authenticationRequest, Handler<AsyncResult<Authentication>> handler);
    
    //invalidate
    void invalidate(AccessKeys accessKeys, Handler<AsyncResult<Void>> handler);
    
    //refresh
    void refresh(RefreshRequest refreshRequest, Handler<AsyncResult<RefreshResponse>> handler);
    
    //siginout
    void siginout(UsernamePassword usernamePassword, Handler<AsyncResult<Void>> handler);
    
    //validate
    void validate(AccessKeys accessKeys, Handler<AsyncResult<Void>> handler);
    
}

File: generated-sources/dart-jaguar/mojang-authentication/lib/api/default_api.dart
import 'package:jaguar_retrofit/annotations/annotations.dart';
import 'package:jaguar_retrofit/jaguar_retrofit.dart';
import 'package:jaguar_serializer/jaguar_serializer.dart';
import 'package:jaguar_serializer/src/repo/repo.dart';
import 'dart:async';

import 'package:openapi/model/authentication_request.dart';
import 'package:openapi/model/refresh_request.dart';
import 'package:openapi/model/refresh_response.dart';
import 'package:openapi/model/username_password.dart';
import 'package:openapi/model/error.dart';
import 'package:openapi/model/access_keys.dart';
import 'package:openapi/model/authentication.dart';


part 'default_api.jretro.dart';

@GenApiClient()
class DefaultApi extends _$DefaultApiClient implements ApiClient {
    final Route base;
    final SerializerRepo serializers;

    DefaultApi({this.base, this.serializers});

    /// 
    ///
    /// 
    @PostReq(path: "/authenticate")
    Future<Authentication> authenticate(
        
        @AsJson() AuthenticationRequest authenticationRequest
    );

    /// 
    ///
    /// 
    @PostReq(path: "/invalidate")
    Future<void> invalidate(
        
        @AsJson() AccessKeys accessKeys
    );

    /// 
    ///
    /// 
    @PostReq(path: "/refresh")
    Future<RefreshResponse> refresh(
        
        @AsJson() RefreshRequest refreshRequest
    );

    /// 
    ///
    /// 
    @PostReq(path: "/signout")
    Future<void> siginout(
        
        @AsJson() UsernamePassword usernamePassword
    );

    /// 
    ///
    /// 
    @PostReq(path: "/validate")
    Future<void> validate(
        
        @AsJson() AccessKeys accessKeys
    );


}
